### DISCUSSION TITLE
Adding custom wordvectors to my pipeline

### DISCUSSION THREAD

**Discussion by Gitclop** [explosion/spaCy#12313] [Help: Coding & Implementations] [answered]
I've build custom wordvectors and created a blank model with 
`spacy init vectors`
Now if i load the model will i loose all the components from the stock models, like de_core_news_sm?
can i use something like 
`nlp.add_pipe('vectors', source=pathToVectors)`
?
Or how would i just load the custom vectors onto my existing pipeline?

**Comment by Gitclop** [comment, marked as answer]
I got it working by iterating the gensim model and adding each vector manually to a new spacy vocab.

```
from gensim.models import Word2Vec
from spacy.vocab import Vocab

gensim_model = Word2Vec.load(my_w2vmodel.model)
vocab = Vocab()

for word in gensim_model.wv.index_to_key:
      vector = gensim_model.wv.get_vector(word)
      vocab.set_vector(word, vector)

nlp.vocab.vectors = vocab.vectors
```

I would like to know if this could cause problems with other pipeline components and what would be the prefered way to load custom wordvectors in spacy.

**Reply by svlandeg** [reply]
Hi! You can indeed either use `spacy init vectors` or `vocab.set_vector`. Whether or not this will influence existing trained components depends on their settings. See the docs [here](https://spacy.io/usage/embeddings-transformers#word-vectors-models) - if `include_static_vectors` is set to `True` and you're changing the vectors after training, then that might negatively influence the model's performance.
